x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/common/DocumentConversionUtils.java

```diff
@@ -11,34 +11,45 @@
 import org.apache.logging.log4j.message.ParameterizedMessage;
 import org.elasticsearch.action.fieldcaps.FieldCapabilitiesResponse;
 import org.elasticsearch.action.index.IndexRequest;
-import org.elasticsearch.xpack.core.transform.TransformField;
 
 import java.util.HashMap;
 import java.util.Map;
 
 import static java.util.function.Predicate.not;
 
+/**
+ * Helper functions for converting raw maps into documents and determining index mappings
+ */
 public class DocumentConversionUtils {
 
     private static final Logger logger = LogManager.getLogger(DocumentConversionUtils.class);
 
-    public static IndexRequest convertDocumentToIndexRequest(Map<String, Object> document,
+    /**
+     * Convert a raw string, object map to a valid index request
+     * @param docId The document ID
+     * @param document The document contents
+     * @param destinationIndex The index where the document is to be indexed
+     * @param destinationPipeline Optional destination pipeline
+     * @return A valid {@link IndexRequest}
+     */
+    public static IndexRequest convertDocumentToIndexRequest(String docId,
```

Comment:
nice re-factoring of the `docId`. If I remember correctly, this removes the need for `removeInternalFields`. I think the only purpose was `_id`. We might get rid of it now.

The idea of having _internal entries_ might still be a good thing, however we can add it back, once we need it (and maybe we pass a real object instead).

So, having that said, feel free to remove `removeInternalFields` (optional).